src/components/map-marker/map-marker.tsx

```diff
@@ -62,16 +69,38 @@ export class MapMarker implements GxComponent {
    */
   @Event() gxMapMarkerDeleted: EventEmitter;
 
+  private setPopup() {
+    const popupHtml = this.element.querySelector("[class='popupHtml']");
+    if (popupHtml.innerHTML) {
+      const marginProportion = 83 / 100;
+      const maxPopupSize = {
+        height:
+          document.querySelector(".gxMap").clientHeight * marginProportion,
+        width: document.querySelector(".gxMap").clientWidth * marginProportion
+      };
+      this.markerInstance.bindPopup(popupHtml, {
+        keepInView: true,
+        maxHeight: maxPopupSize.height,
+        maxWidth: maxPopupSize.width,
+        minWidth: 100
+      });
+    }
+  }
+
   componentDidLoad() {
-    const halfIconWidth = this.iconWidth / 2;
+    const halfIconSizes = {
+      height: this.iconHeight / 2,
+      width: this.iconWidth / 2
+    };
     const coords = parseCoords(this.coords);
     if (coords !== null) {
       this.markerInstance = marker(coords, {
         icon: divIcon({
           className: this.markerClass,
-          iconAnchor: [halfIconWidth, this.iconHeight],
+          iconAnchor: [halfIconSizes.width, this.iconHeight],
+          popupAnchor: [0, -halfIconSizes.height],
           iconSize: [this.iconWidth, this.iconHeight],
-          tooltipAnchor: [0, -28]
+          tooltipAnchor: [0, -halfIconSizes.height]
         })
```

Comment:
I feel there's a refactoring opportunity here. This code block is repeated several times.

Reply:
Yep! I just realize that piece of code is repeated twice in the ```componentDidLoad``` method.
I'll fix that!

Reply:
done in 7dfb8c6 👌